packages/fuchsia_ctl/lib/src/emulator.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+
+import 'operation_result.dart';
+
+/// A wrapper for running Fuchsia images on the Android Emulator (AEMU).
+class Emulator {
+  /// Creates a new wrapper for the `emu` tool.
+  Emulator({
+    @required this.aemuPath,
+    @required this.fuchsiaImagePath,
+    @required this.fuchsiaSdkPath,
+    this.fs = const LocalFileSystem(),
+    this.processManager = const LocalProcessManager(),
+    @required this.qemuKernelPath,
+    this.sshPath = '.fuchsia',
+    @required this.zbiPath,
+  }) : assert(processManager != null);
+
+  /// The path to the AEMU executable on disk.
+  final String aemuPath;
+
+  /// Fuchsia image to load into the emulator.
+  final String fuchsiaImagePath;
+
+  /// The path to the Fuchsia SDK that contains the tools `fvm` and `zbi`.
+  final String fuchsiaSdkPath;
+
+  /// The QEMU kernel image to use. This is only bundled in Fuchsia QEMU images.
+  final String qemuKernelPath;
+
+  /// The path to the directory containing authorized_keys.
+  final String sshPath;
+
+  /// The Fuchsia bootloader image.
+  final String zbiPath;
```

Comment:
I flipped through several iterations of naming this. This is the `zircon-a.zbi` file included in a release image. There is also a tool called `zbi` for working with these in the sdk. I originally had `zirconBootImagePath`. Do you think it would be helpful to use that instead when referring to the image instead of the tool?